packages/liveblocks-core/src/room.ts

```diff
@@ -1868,47 +1862,50 @@ function makeStateMachine<
       context.socket = null;
     }
 
-    updateConnection({ status: "unavailable" }, batchUpdates);
-    clearTimeout(context.timeoutHandles.pongTimeout);
-    if (context.timeoutHandles.flush) {
-      clearTimeout(context.timeoutHandles.flush);
-    }
-    clearTimeout(context.timeoutHandles.reconnect);
-    clearInterval(context.intervalHandles.heartbeat);
-    connect();
-  }
+    clearTimeout(context.timers.flush);
+    clearTimeout(context.timers.reconnect);
+    clearInterval(context.timers.heartbeat);
+    clearTimeout(context.timers.pongTimeout);
 
-  function applyAndSendOps(
-    offlineOps: Map<string, Op>,
-    batchedUpdatesWrapper: (cb: () => void) => void
-  ) {
-    if (offlineOps.size === 0) {
-      return;
-    }
+    batchUpdates(() => {
+      updateConnection({ status: "closed" }, doNotBatchUpdates);
 
-    const messages: ClientMsg<TPresence, TRoomEvent>[] = [];
+      context.others.clearOthers();
+      notify({ others: [{ type: "reset" }] }, doNotBatchUpdates);
+    });
 
-    const ops = Array.from(offlineOps.values());
+    // Clear all event listeners
+    for (const eventSource of Object.values(eventHub)) {
+      eventSource.clear();
+    }
+  }
 
-    const result = applyOps(ops, true);
+  function reconnect() {
+    if (context.socket) {
+      context.socket.removeEventListener("open", onOpen);
+      context.socket.removeEventListener("message", onMessage);
+      context.socket.removeEventListener("close", onClose);
+      context.socket.removeEventListener("error", onError);
+      context.socket.close();
+      context.socket = null;
+    }
 
-    messages.push({
-      type: ClientMsgCode.UPDATE_STORAGE,
-      ops: result.ops,
-    });
+    clearTimeout(context.timers.flush);
+    clearTimeout(context.timers.reconnect);
+    clearInterval(context.timers.heartbeat);
+    clearTimeout(context.timers.pongTimeout);
```

Comment:
same code as in onClose right? Maybe we need a cleanup function or something to abstract that away so they don't diverge.

Reply:
Oh wow yeah. I moved some functions to colocate them, just to get a better overview during my changes. There is no new behavior introduced here. For a better view, it may be more useful to look at each commit individually instead of looking at the final diff.